pyquil/paulis.py

```diff
@@ -323,10 +374,13 @@ def __mul__(self, other):
         :return: A new PauliSum object given by the multiplication.
         :rtype: PauliSum
         """
-        if isinstance(other, PauliTerm):
-            other_terms = [other]
+        if not any([isinstance(other, allowed) for allowed in (Number, PauliTerm, PauliSum)]):
```

Comment:
You should compactify that code:
```any([isinstance(other, allowed) for allowed in (Number, PauliTerm, PauliSum)])```
can be written as
```isinstance(other, (Number, PauliTerm, PauliSum))```

Reply:
:O